LeetCodeProblems/LeetCodeProblems/WordPattern.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LeetCodeProblems
+{
+    class WordPattern
+    {
+        public static bool WordsPattern(string pattern, string s)
+        {
+
+            string[] words = s.Split(' ');
+            if (words.Length != pattern.Length)
+                return false;
+            Dictionary<char, string> Dict = new Dictionary<char, string>();
+            for (int i = 0; i < pattern.Length; i++)
+            {
+                if (Dict.ContainsKey(pattern[i]))
+                {
+                    if (Dict[pattern[i]] != words[i])
+                        return false;
+                }
+                else
+                {
+                    if (Dict.ContainsValue(words[i]))
```

Comment:
this is log(n), you can have another dictionary for the strings, try that (a dic of dic)